### DISCUSSION TITLE
Streaming microphone AND guitar from Garage Band

### DISCUSSION THREAD

**Discussion by andreagrandi** [ExistentialAudio/BlackHole#221] [General]
Hello folks,

I've installed BlackHole today and **tried to follow this tutorial** https://existential.audio/howto/StreamFromGarageBandToGoogleMeet.php
but **unfortunately it doesn't do what I want**. I will now try to be more specific.

The main reason why the above tutorial doesn't work for me is because it only uses one input, while I need to use two and I don't understand how to combine them.

## What I want to achieve

I want to be able to make a Zoom / Skype / Google Meet normal call, but the other person needs to be able to hear both my voice (coming from MacBook Internal Microphone) and my Bass Guitar coming from my sound interface.

### Context 

I want to be able to take bass guitar lessons remotely, so the teacher needs to be able to **hear both my voice and my bass guitar**, and of course I need to be able to hear the teacher in my headphones.

## My Gear and Setup

- MacOS 10.15.6 (Catalina)
- blackhole 0.2.6 (installed from `brew`)
- Steinberg UR22mkII sound interface
- Garage Band
- a bass guitar connected to the sound interface
- a pair of headphones connected to the sound interface

## My usual configuration

I normally have this configuration in Garage Band

![Screenshot 2020-10-28 at 18 53 05](https://user-images.githubusercontent.com/636391/97476650-f4321b00-194e-11eb-9eaa-a2464c1e221d.png)

#### Multi Output Device

<img width="673" alt="Screenshot 2020-10-28 at 20 00 29" src="https://user-images.githubusercontent.com/636391/97483866-475c9b80-1958-11eb-9a16-2941faf58f2e.png">

#### Garage Band

![Screenshot 2020-10-28 at 19 59 44](https://user-images.githubusercontent.com/636391/97483768-272cdc80-1958-11eb-9e69-aef60d6340a4.png)

#### Google Meet

<img width="512" alt="Screenshot 2020-10-28 at 19 58 53" src="https://user-images.githubusercontent.com/636391/97483731-1714fd00-1958-11eb-9e1f-3f0cc7cc2fa5.png">

#### Outcome

I can hear my bass guitar in my headphones (as usual) but Google Meet doesn't get any sound (you see when it gets sound because the icon next to the microphone is animated. I also tried to have a friend joining the call, he couldn't hear any sound nor my voice

### Other configurations

I've also tried other configurations (I haven't recorded any screenshots, but I can take them if you think they may be useful to understand the issue), one where I also created an "Aggregated Device" with both internal microphone and "BlackHole 16ch" and another one with internal microphone and Steinberg / Multi Output, but none was working as expected.

In the best case I got: my friend was able to either listen to my voice or to my bass guitar (depending what I was selecting as Input in Google Meet) but not both. Also, unfortunately I didn't take note of this configuration so I don't remember how to reproduce it.

## Conclusion

**Can anyone please tell me what is the right combination for what I want to achieve?** 

Thanks! 🙏

**Comment by andreagrandi** [comment]
I made some progresses 💪 but I've also found an issue.

First of all I created this "Aggregated Device" and tested it with Zoom (note: **MacBook Pro Microphone is first**)

<img width="1125" alt="Screenshot 2020-10-28 at 21 37 58" src="https://user-images.githubusercontent.com/636391/97494236-55b1b400-1966-11eb-993d-1f7be7ed6ca2.png">

Zoom could detect my voice but not (for example) a song playing on Spotify (my system settings are Output --> Multi Output Device).

Then I tried to change the order of inputs in "Aggregated Device" like this (note: **BlackHole 16ch is first**)

<img width="1111" alt="Screenshot 2020-10-28 at 21 39 26" src="https://user-images.githubusercontent.com/636391/97494402-8bef3380-1966-11eb-81f4-97edeaa480db.png">

This time Zoom cannot detect my voice, but it can detect the Spotify song.

If I select "BlackHole 16ch" as device in Zoom, it can detect the song from Spotify, but not my voice:

<img width="558" alt="Screenshot 2020-10-28 at 21 47 43" src="https://user-images.githubusercontent.com/636391/97494959-5b5bc980-1967-11eb-875c-d7552a96b4a9.png">


**Outcome:** the only detected device is the first one in the list. How do I make them **both** working at the same time?

**Comment by devinroth** [comment]
You need to set BlackHole as your input into Zoom.

Then using a different program, GarageBand for example, mix your mic/input into BlackHole.

You can follow the steps here.

https://existential.audio/howto/StreamFromGarageBandToZoom.php

**Comment by andreagrandi** [comment]
Hi and thanks for your reply. As I mentioned at the beginning of my message, Garage Band is unable to mix **both** MacBook Microphone and the input from my bass Guitar.

When you select the input device in Garage Band you can only select one device: if you select the microphone (like those instructions shows) you can't select Steinberg sound card and viceversa.

**Comment by devinroth** [comment]
Well then you need some other sort of DAW or something similar.

**Comment by andreagrandi** [comment]
Do you think it could work if I simply connected another microphone to input 1 of my Steinberg sound card while bass guitar is connected to input 2? This way I could use a single input source

**Comment by devinroth** [comment]
Maybe. Worth trying.

Did you already use an aggregate for the input into GarageBand?

**Comment by andreagrandi** [comment]
No I haven't tried this. I will try it tomorrow (I don't have the laptop with me right now) but I suspect it will behave like Zoom before: the highest entry in "Aggregate Device" has priority and it's the only one being redirected. I will give it a try anyway and let you know, thanks.

**Comment by devinroth** [comment]
No. 

Aggregate into to Garage Band.

BlackHole into Zoom.

**Comment by andreagrandi** [comment]
Hi @devinroth thanks for your advice yesterday, I made some progresses but I'm not done yet: now I can make Garage Band to record both from MacBook Mic and my bass guitar (Steinberg interface) at the same time but Zoom / Meet only hear the bass guitar.

This is my current configuration

### Aggregate Device

<img width="674" alt="Screenshot 2020-10-29 at 10 49 31" src="https://user-images.githubusercontent.com/636391/97553347-9d235900-19d5-11eb-986b-77b45a864844.png">

### Garage Band - Options

![Screenshot 2020-10-29 at 10 55 49](https://user-images.githubusercontent.com/636391/97553397-ab717500-19d5-11eb-8a2e-4497e54ef99a.png)

### Garage Band - Tracks

![Screenshot 2020-10-29 at 10 49 52](https://user-images.githubusercontent.com/636391/97553483-c7751680-19d5-11eb-88b2-b166f81b995c.png)

- Track 1 is the bass guitar, configured on **Input 2**
- Track 2 is the MacBook Mic configured on **Input 3**

### Google Meet

<img width="491" alt="Screenshot 2020-10-29 at 10 55 28" src="https://user-images.githubusercontent.com/636391/97553605-effd1080-19d5-11eb-81db-ed509d505055.png">

### Outcome

Google Meet (or Zoom with same configuration) can only hear the bass guitar sound and not my voice

### Question

I've the feeling I'm getting really close to the solution and I'm probably only missing a minor thing to make everything work. Do you have any idea what I may be missing, please? Thanks again for your help

**Comment by andreagrandi** [comment]
A curious thing: if with the same **Google Meet** configuration I click on **Test**, the microphone icon above detects the sound

<img width="794" alt="Screenshot 2020-10-29 at 11 06 46" src="https://user-images.githubusercontent.com/636391/97554446-12435e00-19d7-11eb-9ae0-3ad90191e9b6.png">

**Comment by andreagrandi** [comment]
So, I just discovered another thing and I have also done a test using an external microphone, connected to Line 1 of my Steinberg.

Both using the internal MacBook mic or the external one, the problem is the same: I can't hear the voice in Zoom / Youtube / Skype etc.... but I'm able to record it in Garage Band.

If I activate the **MONITORING** (the icon that looks like an upside down WIFI signal) for the microphone track, I can hear it on Zoom / Skype / Meet etc...

Now: if I activate the monitoring on the external microphone, it's all fine, but if I do it on internal one I hear a noise which will also go in the final recording.

Last but not least: I've tried to record a "Live" video on Youtube and regardless if I include the microphone or not (I only include music from Spotify and bass guitar) the quality is very bad (if you want I can link you a video where I play... but I warn you: I don't play very well :P )

What else am I missing?

**Comment by andreagrandi** [comment]
p.s: at this point I'm also tempted to try the "2ch" version, but I'm not sure what are the differences between the two versions.

**Comment by devinroth** [comment]
You need to enable monitoring (orange button) on both tracks.

**Comment by andreagrandi** [comment]
> You need to enable monitoring (orange button) on both tracks.

Thanks. Enabling monitoring on both tracks it works, but if I use "MacBook Pro Mic" there is a lot of noise (which is not present if I use an external pro microphone). My final workaround to make everything work will be to buy an external microphone I can connect to my Steinberg.

Thanks again for your help. Once I've sorted out everything I will probably blog a complete tutorial, because I know a lot of people who may have the exact same requirement.

**Reply by silvioalvim** [reply]
I managed to streaming both tracks with my Scarlett 2i2 Interface, with guitar and mic, but with feedback... i have to use headphones

**Comment by devinroth** [comment]
Awesome.

**Reply by erinhyvin** [reply]
Hi all, 
I followed all these steps and I can't find what is happening. 
I am trying to go on google meet and be able to sing and play guitar. I have a microphone and guitar plugged into Garage Band with an interface (Scarlett) and I follow all the steps and no sounds comes up on google Hangouts. I am attaching screenshots of every step, can you please tell me what am I doing wrong?


1. Configurated my settings on Blackhole

![Captura de pantalla 2021-03-14 a las 10 46 39 PM](https://user-images.githubusercontent.com/80650496/111085413-983feb80-8517-11eb-9b4e-5284aa2cd983.png)


2. Modified the audio settings on Garage Band

![Captura de pantalla 2021-03-14 a las 10 48 06 PM](https://user-images.githubusercontent.com/80650496/111085434-aaba2500-8517-11eb-856f-5b3c6ab2ee4a.png)

3. Started a google meet session and after modifying the settings to the ones you see in the pic below, i can hear the sound on my headphones but the person in the meeting can't. (By the way as my speakers I have Multi-Output Device)
![Captura de pantalla 2021-03-14 a las 10 49 05 PM](https://user-images.githubusercontent.com/80650496/111085487-d63d0f80-8517-11eb-9f1d-badecbb4640c.png)

Can anybody please help? I really need to solve this as I have an audition this week. 
Thank you!

**Comment by devinroth** [comment]
You need a 2ch device as the top device in a multi output.

https://github.com/ExistentialAudio/BlackHole/wiki/Multi-Output-Device#4-select-output-devices

Devin Roth

**Reply by erinhyvin** [reply]
Hi Devin,
What is a 2ch device can you please show me a picture im pretty new at this. Are the rest of my settings right? Please let me know. Thanks

**Reply by alexbbrown** [reply]
the 2ch (or 16ch, depending how you have configured it) is the BlackHole device.

**Comment by devinroth** [comment]
Did you even look at the link I sent?

Devin Roth

**Reply by erinhyvin** [reply]
Hi Devin, 
Sorry about that. Yes, I read everything again as I already saw that page before and still not working. I show you my settings: 

![Captura de pantalla 2021-03-15 a las 10 05 43 PM](https://user-images.githubusercontent.com/80650496/111221337-a571de80-85da-11eb-8c8a-5e8bafc5c447.png)

![2](https://user-images.githubusercontent.com/80650496/111221352-ac98ec80-85da-11eb-9a2c-8bc59341899f.png)


![3](https://user-images.githubusercontent.com/80650496/111221379-b6225480-85da-11eb-822c-72c254f190dd.png)


Do you know what am i doing wrong? sorry cause since English is not my first language maybe I'm missing something. 

thank you!

**Reply by alexbbrown** [reply]
Your multi-output device needs to have the blackhole 16-ch (aka 2ch) device at the top.
